tempesta_fw/cache.c

```diff
@@ -1210,56 +1204,81 @@ cache_req_process_node(TfwHttpReq *req, unsigned long key,
 	tfw_cache_dbce_put(ce);
 }
 
-/*
- * Invalidate a cache entry.
- * In fact, this is implemented by making the cache entry stale.
- */
-static int
-tfw_cache_purge_invalidate(TfwHttpReq *req, unsigned long key)
+static void
+tfw_cache_do_action(TfwHttpReq *req, TfwHttpResp *resp,
+		    tfw_http_cache_cb_t action)
 {
-	TdbIter iter;
-	TDB *db = node_db();
-	TfwCacheEntry *ce = NULL;
+	if (resp) {
+		tfw_cache_add(resp, req, action);
+	}
+	else if (req->method == TFW_HTTP_METH_PURGE) {
+		tfw_cache_purge_method(req);
+	}
+	else {
+		cache_req_process_node(req, action);
+	}
```

Comment:
Usually we use curly braces in comples if-else statements: https://github.com/tempesta-tech/linux-4.1.27-tfw/blob/master/Documentation/CodingStyle#L141